colcon_core/command.py

```diff
@@ -86,7 +86,6 @@ def register_command_exit_handler(handler):
 
     :param handler: The callable
     """
-    global _command_exit_handlers
```

Comment:
I don't think is trivial removing global in relation to `_command_exit_handlers` that variable is defined as a global variable in line https://github.com/colcon/colcon-core/blob/03d619313015cd90768c24a5347c8664c4b2dc13/colcon_core/command.py#L78. 
The prefix `global` in python is use to be able to read and write a global variable in the context of a specific function. Without that prefix whatever change you do to that variable within the function won't take effect on the global defined variable but a copy of it.

Reply:
> Without that prefix whatever change you do to that variable within the function won't take effect on the global defined variable but a copy of it.

I don't think this statement is true as written. The `global` statement makes value replacement operations affect the global variable instead of creating a local variable, but MODIFYING an object behind a global variable is not affected by the `global` statement at all, and has always been the implicit behavior.

```python
>>> foo = []
>>> def doit():
...   foo.append('hello, world')
... 
>>> doit()
>>> foo
['hello, world']
>>>
```

Reply:
With a replacement-type operation, the `global` statement is needed:
```python
>>> foo = 0
>>> def doit():
...   foo = 1
... 
>>> doit()
>>> foo
0
>>> def doit_better():
...   global foo
...   foo = 1
... 
>>> doit_better()
>>> foo
1
>>>
```

Reply:
It seems I have been coding immutable code for long enough that I have forgotten that a lot of data structures are mutable. For the use case of this code, you are right and as long as we are not doing value replacements, there is no explicit need of the global prefix. 
I would even argue in favor of not having the global prefix, since it removes the effect of replacing the value of the global variable and could prevent some unexpected behavior (aka someone adds a line like `foo =['this won't do anything']`).